src/server/proxy_protocol.rs

```diff
@@ -35,20 +36,13 @@ pub enum ProxyError {
 
 #[derive(Debug, Copy, Clone)]
 pub struct ConnectionTuple {
-    pub from_ip: IpAddr,
-    pub from_port: u16,
-    pub to_ip: IpAddr,
-    pub to_port: u16,
+    pub source: SocketAddr,
+    pub destination: SocketAddr,
 }
 
 impl ConnectionTuple {
-    fn new(from_ip: IpAddr, from_port: u16, to_ip: IpAddr, to_port: u16) -> Self {
-        ConnectionTuple {
-            from_ip,
-            from_port,
-            to_ip,
-            to_port,
-        }
+    pub fn hash(&self) -> String {
```

Comment:
A side comment for another day: This is hardly a `hash`.  Suggest using the name `key` rather as method name.

Reply:
renamed